oak_io/src/lib.rs

```diff
@@ -33,12 +33,22 @@ pub use receiver::Receiver;
 pub use sender::Sender;
 
 /// A simple holder for bytes + handles, using internally owned buffers.
-#[derive(Clone, Debug, PartialEq, Eq)]
+#[derive(Clone, PartialEq, Eq)]
 pub struct Message {
     pub bytes: Vec<u8>,
     pub handles: Vec<Handle>,
 }
 
+/// Manual `Debug` implementation to allow hex display.
+impl std::fmt::Debug for Message {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("Message")
+            .field("handles", &self.handles)
+            .field("bytes", &format_args!("{}", hex::encode(&self.bytes)))
```

Comment:
I guess just using `{:x?}` does not do what you want?

Reply:
Sadly no, it still formats like `[0a, 12, 5f, 0b, 7]`